content/docs/use-cases/datasets-cache.md

```diff
@@ -0,0 +1,107 @@
+# Datasets Cache
```

Comment:
Why "datasets" instead of "data"? To follow the structure of "Versioning Data and Models" and "Sharing Data and Model Files", what about "Caching Data and Models"?

Reply:
I can give my 2cs on the naming (and maybe we can brainstorm how to make to more explicit by changing the title and/or improving the first paragraph a bit). This is indeed the most tricky and the most challenging part of the use cases usually :)

My take on this use case is that by making DVC cache and shared global we are making it an explicit infrastructure element (along with training machines or cluster, along with regular storage that is usually S3 or something, etc). It gives clear benefits, goals, lifecycle, connections to other infrastructure, etc. In this sense, it should be more about organizing infrastructure, then workflow. 

That's why probably names like "Data Cache Layer", "Datasets Cache" (pretty much the same, but domain-specific) resonate a bit better with me than "Caching Data and Models". The latter for me also covers the simple project level cache for example. And may be that's also fine, btw - we can expand and cover the local project cache as one way to do this! 

I hope it makes at least some sense. Happy to explore this together :)

Reply:
> Why "datasets" instead of "data"?

Yeah probably "Data" is better @dberenbaum. The trickier part is the term "cache" I think, especially readers already immersed in DVC concepts (like us). Try to read this from the PoV of a new user. "Caching" has a general meaning (not necessarily DVC cache), i.e. an intermediate storage layer that's faster but limited (smaller, more expensive). Regardless, I'll try to disambiguate for existing users (same applies for term "remote").

Also, I brainstormed on this with @iesahin and some other ideas we had are: Common Data Cache, Frequent-access Cache, Fast Data Access, Multiple Storage Layers. An interesting principle to include (maybe implicitly): "20% data used 80% of the time"

> it should be more about organizing infrastructure, then workflow

Yes I am also improving on this, to make it clear that we are presenting a pattern in this use case, not a specific DVC project setup (again tricky due to ambiguous terms).

> "Caching Data and Models" also covers the simple project level cache for example... we can expand and cover the local project cache as one way to do this!

Let's pin that for now 😅

Reply:
Yup, I know a lot of thought was put into it, which was why I felt like it was necessary to comment. Could just be my bias. Will be interested to hear opinions of others.

Reply:
Went for **Fast Data Storage Layer** for now 😋

Reply:
p.s. @shcheklein I think naming and intro, as well as improving the figures are the 2 major things missing. PTAL

Reply:
What about "Joint Cache Layer" or "Joint Storage Layer" for the name?

Reply:
What about "Joint Cache" or "Joint Data Layer"?

Reply:
Intermediate/Near Cache?

Reply:
"Joint" typically refers to 2 things (e.g. 2 parties working together). So still no perfect title... 🤷

Reply:
"Intermediate" and "Near" can be implied by "Cache", I think. It's a tricky one

Reply:
I think "fast" is more misleading than "joint." It emphasizes the speed. We don't know if their connection in LAN is _that_ faster than the cloud. 

I think "joint" having 2 parties is OK. We are discussing to share the cache in a LAN and this implies there are at least 2 parties. 

Some more: 

- Local Shared Cache
- Local Joint Cache
- Local Common Cache
- Local Intermediate Cache
- Warm Cache (vs. Cold Cache)
- Neighbor(ing) Cache (vs. Remote Cache)
- Converging Cache (vs. Diverging Cache)
- Nearby Cache
- Adjacent Cache
- Adjoint Cache (vs. Disjoint Remote / Cache)

Reply:
I think we can describe hot / warm / cold cache layers, hot being local to machine, warm being within the same LAN, cold is in the cloud.

Reply:
> "fast" is more misleading than "joint." It emphasizes the speed.

Yep, that's the emphasis we want to give! I think it's OK if we are clear in the content that the speed comes from making an intermediate storage layer that is near the local env (in the local network or VPN for example).

> I think "joint" having 2 parties is OK.

Joint usually refers to exactly 2 things working on something specific together. What are the 2 things and what are they doing together? Maybe I just don't understand your idea.

> Warm/Neighboring/Near/Adjacent

Those terms could be 👍 

> hot / warm / cold cache layers

I don't want to call everything "cache". We're keeping it more general with the term "storage". Remote storage is not a cache for example. And what would "hot cache" be? There's just one cache I think (the "warm" one). Local storage is not even needed (links are used).

Reply:
> Joint usually refers to exactly 2 things working on something specific together. What are the 2 things and what are they doing together? Maybe I just don't understand your idea.

I'm looking at the definition and doesn't see "exactly two". Adjective definition says "two or more". 

![2021-07-31 (2)](https://user-images.githubusercontent.com/476310/127734322-24626b2e-a19f-482d-81af-0d3149d36c72.png)

We have two or more systems sharing a single ("near", "fast", whatever) cache or storage. "Joint" means "shared" in this context.

Reply:
You're right, it's 2+ as an adjective. But IMO "joint" is still not great because it refers to things working together as one and here we're separating a storage patter into layers. Each one is independent to some degree, although DVC makes them work together but that's not the main point. They don't work together per se.

I'll summarize names in a separate comment for everyone to weight in.

Reply:
https://github.com/iterative/dvc.org/pull/2587#issuecomment-892240166